controllers/user.controller.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const filterProps = require('../services/utils.js').filterProps;
+
+class UsersController {
+  constructor (userModel) {
+    if (!userModel) throw new Error('User model not provided');
+    this.User = userModel;
+    this.createUser = this.createUser.bind(this);
+    this.signIn = this.signIn.bind(this);
+    this.getUser = this.getUser.bind(this);
+  }
+
+  async createUser (ctx, next) {
+    if (ctx.method !== 'POST') throw new Error('Method not allowed');
+    const userData = ctx.request.body;
+    if (userData.email && userData.password) {
+      let user = await this.User.findOne({
+        where: {
+          email: userData.email,
+        }
+      });
+      if (user) {
+        ctx.status = 401;
+        ctx.body = {
+          errors: ['User already exists.']
+        };
+      } else {
+        user = filterProps(userData, ['email', 'first_name', 'last_name']);
+        user.hash_password = await bcrypt.hash(userData.password, 10);
+        let newUser = await this.User.create(user);
+        ctx.body = filterProps(newUser.dataValues, ['id', 'email', 'first_name', 'last_name']);
+        ctx.status = 201;
+      }
+    } else {
+      ctx.status = 406;
+      ctx.body = {
+        errors: ['Email and/or password needed']
+      };
+    }
+  }
+
+  async signIn (ctx, next) {
+    if (ctx.method !== 'POST') throw new Error('Method not allowed');
+    const body = ctx.request.body;
+    const user = await this.User.findOne({
+      where: {
+        email: body.email,
```

Comment:
Don't use the body, use the `authorization` header.

Reply:
Ok we talked about it, but just to keep record of the conversation.
This is the sign in, no token required yet. This should follow Basic Authentication and the authorization data should be sent through the `Authorization` Header.